modules/core/src/main/java/org/apache/synapse/config/xml/MessageStoreMediatorSerializer.java

```diff
@@ -62,7 +64,9 @@ protected OMElement serializeSpecificMediator(Mediator m) {
 
         //In normal operations messageStoreName can't be null
         //But we do a null check here since in run time there can be manuel modifications
-        if(messageStoreName != null ) {
+        if(messageStoreExp != null) {
```

Comment:
Code is not formatted correctly, please fix it

Reply:
fixed